web/modules/custom/joinup_core/src/Plugin/Field/FieldFormatter/EntityReferenceCustomLinkFormatter.php

```diff
@@ -0,0 +1,235 @@
+    ] + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $elements['path'] = [
+      '#title' => $this->t('The path of the link'),
+      '#description' => $this->t('The path supports tokens e.g. [term:id]. Internal paths should start with "internal:/"'),
+      '#type' => 'textfield',
+      '#required' => TRUE,
+      '#default_value' => $this->getSetting('path'),
+    ];
+
+    $elements['label'] = [
+      '#title' => $this->t('The link text to display'),
+      '#description' => $this->t('The text supports tokens e.g. [term:label].'),
+      '#type' => 'textfield',
+      '#required' => TRUE,
+      '#default_value' => $this->getSetting('label'),
+    ];
+
+    $elements['query_parameters'] = [
+      '#title' => $this->t('Query parameters'),
+      '#description' => $this->t('Add one query parameter per line split by "|". The text supports tokens e.g. [term:label]. The query parameter cannot contain the | symbol.'),
+      '#type' => 'textarea',
+      '#default_value' => $this->getSetting('query_parameters'),
+    ];
+
+    $elements['limit'] = [
+      '#title' => $this->t('Maximum amount of links to show.'),
+      '#description' => $this->t('Set this to -1 to display all links.'),
+      '#attributes' => [
+        ' type' => 'number',
+      ],
+      '#required' => TRUE,
+      '#type' => 'textfield',
+      '#default_value' => $this->getSetting('limit'),
+    ];
+
+    return $elements;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsSummary() {
+    $summary = [];
+    $summary[] = $this->getSetting('path') ?
+      $this->t('Redirect path: %path.', [
+        '%path' => $this->getSetting('path'),
+      ]) : $this->t('No link');
+    $summary[] = $this->t('Link label: %label.', [
+      '%label' => $this->getSetting('label'),
+    ]);
+    $summary[] = $this->t('Query parameters: %parameters.', [
+      '%parameters' => $this->getSetting('query_parameters'),
+    ]);
+    $summary[] = $this->t('Limit: %limit', [
+      '%limit' => $this->getSetting('limit'),
+    ]);
+
+    return $summary;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $elements = [];
+    $entities_to_view = $this->getEntitiesToView($items, $langcode);
+    $limit = $this->getSetting('limit');
+    if ($limit !== -1 && $limit < count($entities_to_view)) {
+      // Due to the fact that delta is not yet implemented in SPARQL, in case
+      // not all terms are shown, sort the terms alphabetically. This will keep
+      // the results predictable and consistent across the website.
+      usort($entities_to_view, function (EntityInterface $entity_a, EntityInterface $entity_b): int {
+        return $entity_a->label() <=> $entity_b->label();
+      });
+      $entities_to_view = array_splice($entities_to_view, 0, (int) $this->getSetting('limit'));
+    }
```

Comment:
This part makes this formatter so custom, so "only Joinup thing" that I don't see the reason why it was made so configurable.

Also, conceptually, `$this->getEntitiesToView()` should be in business to perform these operations of limiting and sorting, as its name pretends.

Reply:
It is mostly configurable because you guys like things to be configurable in the UI and stored in YAML files instead of hardcoding logic here and there.
Indeed this part was a bit unavoidable because tests would not be consistent (I could not get CPHP to work) which means the resulting limited array would also not be consistent and predictable. 
Of course there was a simpler way to test this by simply checking 7 links, but I thought that the predictability was also an issue.

Apart from that, as we discussed in the chat - adding here for future reference - I understand that this whole formatter is specific, but the formatter itself is mainly allowing the user to use tokens and control the output as link. I have put in place some limitations because they cover our use case now. In the future this can be extended or have a base class which this class will extend. The configurability will come almost out of the box. This for example can be a class that in the future extends a "Custom text formatter" which is not a link but still allows the user to output the field as a custom token supporting text.